### ISSUE TITLE
Area2D: error displayed on console but not in editor

### ISSUE THREAD

**Issue by cbscribe** [godotengine/godot#8602]
EDIT: Updated title to reflect that this issue is really due to the following error:

    ERROR: Function blocked during in/out signal. Use call_deferred("set_enable_monitoring",true/false)
   At: scene\2d\area_2d.cpp:429

However, the error does not appear when running the project in the Godot editor.  The error *does* appear when running the project from the command line.

**Operating system or device - Godot version:**
OS X/ Linux - 2.1.2/2.1.3

**Issue description:**
Area2D node with body_enter signal connected to a function.  Monitoring cannot be disabled in that function.

**Steps to reproduce:**
Connect the signal and create a function:

    func _ready():
        connect("body_enter", self, "_on_body_enter")

    func _on_body_enter( body ):
        set_enable_monitoring(false)
        print(is_monitoring_enabled())

Result is `True'

Placing those two lines in any other function (`_ready()` or `_process()` for example) works as expected - 'False' is displayed.

**Link to minimal example project:**
Example - 1 moving body & one area2d.
[area2d_test.zip](https://github.com/godotengine/godot/files/967173/area2d_test.zip)

**Comment by eon-s** [comment]
It throws an error if you run the editor from a terminal:

```
ERROR: Function blocked during in/out signal. Use call_deferred("set_enable_monitoring",true/false)
   At: scene\2d\area_2d.cpp:429
```

The bug here is that the error description is empty in the debugger...

**Comment by cbscribe** [comment]
Nice catch!  I didn't even consider that the output console would be different from command line.  Definitely a problem - even if it's not the original one I thought.

**Comment by eon-s** [comment]
@cbscribe yeah, realized that was a call_deferred situation when saw `Condition 'locked' is true` in the debugger and then looked at the terminal, sometimes it gives more information (sometimes don't, like when using environments). 
I hope that kind of mysterious errors get fixed on 3 at least.

**Comment by karroffel** [comment]
Can this be closed then?

**Comment by magyar123** [comment]
I think this reveals a slightly bigger problem, that some of the errors are only displayed on the console, and some are logged in the editor log too.

**Comment by karroffel** [comment]
Then open a separate issue for that or change the issue in such a way that it's obvious what this discussion is about

**Comment by kubecz3k** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about its current status here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by eon-s** [comment]
This is already solved on 2.1 and 3 branches, the error can be seen in the debugger.
